### ISSUE TITLE
"Trust disables VPN" may break Facetime while connected to VPN

### ISSUE THREAD

**Issue by seanob86** [passepartoutvpn/passepartout-ios#98]
First off - I don't know if this is an Apple iOS issue rather than an issue with Passepartout app. Also, I don't know if this issue was recently introduced (if by iOS update or Passepartout update). It's an issue that I have come across by chance as I rely on Facetime to contact family/friends while I live overseas.

Passepartout v. 1.8.0 (1982)
iOS 12.4.1 on iPhone X

After discovering Facetime was not notifying me of an incoming call I tested until I found what needed to happen to break Facetime. **A sequence of events** need to take place in order to break Facetime. Once Facetime is broken incoming and outgoing calls fail to make a connection to the receiver device. I use Passepartout 24/7 and allow the `Trust disables VPN` setting to handle when the VPN should disconnect/connect, however from my testing it seems that care should be taken when disabling the WiFi.

Unfortunately once Facetime is broken (while connected to VPN), the only way you will know it's broken (without actively checking it) is if you are unable to make an outgoing call or someone tells you that they tried calling you.

I am using the built-in providers to initiate a VPN connection. I am able to successfully break Facetime with either `Mullvad` or `NordVPN`.

**Expected Result**
Both incoming and outgoing calls are received (while connected to VPN).

**Actual Result**
Both incoming and outgoing calls are **not** received (while connected to VPN).

**Steps to reproduce the issue**
**Note:** To test you will need another apple device signed in to FaceTime with another iCloud/Apple ID account so that you may Facetime call device with Passepartout app to reproduce issue.

1. Ensure there are no active VPN connections on an iPhone with Passepartout installed. Create a `trusted network` (to be able to trust a Wi-Fi network you should already be connected to that network with WiFi enabled). Toggle the newly added trusted network **ON**. In my setup my home WiFi is my only trusted network. `Cellular network` is **not** trusted.

2. **Enable** `Trust disables VPN `and `Keep alive on sleep`. **Do not** enable the VPN in the app at this point.

3. Disable the device WiFi either via settings or control center. Then in the app enable the VPN to allow the VPN to connect (as you should be on an untrusted Network i.e. Cellular).  Facetime is still operational at this point. 

4. Enable the device WiFi to allow iPhone to connect to trusted WiFi. Confirm the VPN disconnects automatically. Then shutdown the WiFi again to fallback to untrusted Cellular network. Confirm VPN reconnects.

5. This is where you need another device to help test Facetime. Test Facetime on your iPhone (where Passepartout is installed) by: 
          A) receiving a Facetime call and 
          B) Initiating a Facetime call 

Facetime should not be operational on the iPhone now **while the VPN is connected**. Outgoing calls are not received by the destination device and your iPhone is unable to receive Facetime calls.

At this point even by simply reconnecting to the VPN, Facetime will still be broken (while VPN is connected) and the only way to recover Facetime is by either rebooting the device or by the option `Remove VPN configuration` via the app settings (and then installing the configuration by enabling the VPN)

In short - with the `Trust disables VPN` option enabled this issue may be experienced if the user disables the device WiFi (for whatever reason) which could trigger the VPN to either disconnect/connect depending on their trusted networks.

**Troubleshooting**

- Reset iPhone Network Settings via Settings, General, Reset
- Reinstall Passepartout
- Install Passepartout Beta 1.8.1 (2002)
- With `Trust disables VPN` **disabled** I have performed the same sequence of events where I manually disconnect/reconnect VPN (where it would otherwise be handled automatically). This manual intervention of disabling/enabling VPN does not break Facetime. 
- Allowing the device to roam in and out of the trusted network physically does not seem to break Facetime i.e. taking the device in and out of range of the trusted network to disconnect/connect to VPN.
- iOS 13? Unfortunately, I do not have a device to test with.
- I am unable to test where the VPN tunnel unexpectedly disconnects/connects.

**Comment by keeshux** [comment]
@seanob86 updates on this? Please be more on point, the issue description is quite chaotic.

**Comment by seanob86** [comment]
Yeah sorry about the chaotic description. I was trying to be as descriptive as possible to  help reproduce the problem. From memory (as it was a little while ago) incoming and outgoing FaceTime calls will not ring on the other end whilst connected to a VPN (regardless of vendor/config). On iOS 12.x (whatever was the latest version prior to iOS 13) I could easily reproduce the issue.

Good news iOS 13 solved the issue. Out of frustration (when I raised this issue) I updated to the iOS 13 GM and that instantly resolved this. I also found that WireGuard app also had the same issue on iOS 12. So if anything it’s iOS related. I don’t know if the issue is still present on iOS 12 but from my perspective I don’t care anymore as iOS 13 has resolved this. 

Ok for you to close if you see fit as I’m sure you have more pressing issues/feature updates that are more important. 

Actually I’ll do one better and close this myself 😊
